lib/compiler.js

```diff
@@ -670,51 +699,52 @@ function fixFileExtension(filepath, files, onlyScriptsExportingFunction) {
 
       if (!(otherFileExtension in getExcludedExtensions()) &&
         path.basename(f, otherFileExtension) == basename) {
-        if (!onlyScriptsExportingFunction)
-          results.push(otherFile);
-        else if (onlyScriptsExportingFunction &&
-          (typeof require.extensions[otherFileExtension]) === 'function') {
-          results.push(otherFile);
-        }
+        // if (!onlyScriptsExportingFunction)
+        results.push(otherFile);
+        //else if (onlyScriptsExportingFunction &&
+          //(typeof scriptExtensions[otherFileExtension]) === 'function') {
+          //results.push(otherFile);
+        //}
       }
     }
   });
   return (results.length > 0 ? results[0] : undefined);
 }
 
-function resolveAppScriptPath(rootDir, relativePath, resolveOptions) {
+function resolveAppScriptPath(rootDir, relativePath, resolveOptions,
+                              scriptExtensions) {
   var resolvedPath = resolveAppPath(rootDir, relativePath, resolveOptions);
   if (!resolvedPath) {
     return false;
   }
   var sourceDir = path.dirname(resolvedPath);
   var files = tryReadDir(sourceDir);
-  var fixedFile = fixFileExtension(resolvedPath, files, false);
+  var fixedFile = fixFileExtension(resolvedPath, files, false,
+                                   scriptExtensions);
   return (fixedFile === undefined ? resolvedPath : fixedFile);
 }
 
-function buildAllMixinInstructions(appRootDir, mixinDirs, mixinSources, options,
-  modelInstructions) {
-  var extensions = _.without(_.keys(require.extensions),
-    _.keys(getExcludedExtensions()));
-
+function buildAllMixinInstructions(appRootDir, mixins, mixinDirs, mixinSources,
+                                   scriptExtensions, normalization,
+                                   modelInstructions) {
   // load mixins from `options.mixins`
-  var sourceFiles = options.mixins || [];
```

Comment:
What is the rationale for this change? I mean why to pass `options` properties as individual arguments? `buildAllMixinInstructions` is already accepting way too many arguments, we should look into ways how to make that list shorter, not longer...